docker-compose.yml

```diff
@@ -0,0 +1,26 @@
+version: '3.8'
+services:
+  app:
+    build:
+      context: .
+      args:
+        NODE_ENV: development
+    volumes:
+      - node_modules:/app/node_modules
```

Comment:
we are mounting a named volume "node_modules" at the path in the container where we expect it to be. This is shared between the container and the host, but is actually independent of both, managed by docker and this is how we avoid the permissions conflict because both parties are given access by the real owner.